tests/Resources/ti.blob.test.js

```diff
@@ -0,0 +1,337 @@
+/*
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2011-Present by Appcelerator, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+/* eslint-env mocha */
+/* global Ti */
+/* eslint no-unused-expressions: "off" */
+'use strict';
+var should = require('./utilities/assertions'),
+	utilities = require('./utilities/utilities');
+
+describe.only('Titanium.Blob', function () {
+	var win;
+
+	afterEach(function () {
+		if (win) {
+			win.close();
+		}
+		win = null;
+	});
+
+	it('.apiName', function () {
+		// FIXME Should be able to do Ti.Blob.apiName
+		var blob = Ti.Filesystem.getFile('app.js').read();
+		should(blob).have.a.readOnlyProperty('apiName').which.is.a.String;
+		should(blob.apiName).be.eql('Ti.Blob');
+	});
+
+	it('constructed from File.read()', function () {
+		var blob = Ti.Filesystem.getFile('app.js').read();
+		should(blob).be.an.Object;
+		// should(blob).be.an.instanceof(Ti.Blob); // Crashes Windows, throws uncaught error on iOS & Android
+	});
+
+	// Windows crashes on instanceof check TIMOB-25012
+	// Windows also crashes if we uncomment this now, I think closing the window (or failing the test) in the blob callback is causing Desktop crash
+	// Android is sometimes timing out... Trying an open event now...
+	// TODO: Test is tempermental, skipping for now...
+	it('constructed from image', function (finish) {
+		var label;
+		win = Ti.UI.createWindow();
+		label = Ti.UI.createLabel({ text: 'test' });
+		win.add(label);
+		win.addEventListener('open', function () {
+			label.toImage(function (blob) {
+				should(blob).be.an.Object;
+				// should(blob).be.an.instanceof(Ti.Blob); // FIXME Crashes Windows, throws uncaught error on iOS & Android
+				// should(blob.getText()).equal(null); // FIXME 'blob.getText is not a function' on iOS
+				// should(blob.text).equal(null); // FIXME this is undefined on iOS, docs say it should be null
+				should(blob.text).not.exist;
+				Ti.API.info(blob.width);
+				should(blob.width).be.a.Number; // FIXME Undefined on iOS
+				should(blob.width).be.above(0); // 0 on Windows
+				should(blob.height).be.a.Number;
+				should(blob.height).be.above(0);
+				should(blob.length).be.a.Number;
+				// FIXME Parity issue, no size property on Android
+				if (!utilities.isAndroid()) {
```

Comment:
Could this be addressed as part of this PR? I think we receive the size internally already, so a simple `@Kroll.getProperty` may be able to clear this.

Reply:
Oh, you exposed it already :-). Then the guard can be removed.